packages/documentation-site/patternfly-docs/content/accessibility/tree-view/tree-view.md

```diff
@@ -3,32 +3,89 @@ id: Tree view
+For the HTML/CSS library:
 
-The following props/attributes have been added for you or are customizable in PatternFly:
+- Ensure only the root `ul.pf-v6-c-tree-view__list` element has the attribute `role="tree"`, and that any nested `ul.pf-v6-c-tree-view__list` elements instead have `role="group"`.
 
-| React prop | React component that it should be applied to | Which HTML element it appears on in markup | Reason used |
-| -- | -- | -- | -- |
-| `checkProps` | TreeViewListItem | checkbox input | Additional properties of the tree view item checkbox; defaulted to `{ checked: false }` |
-| `searchProps` | TreeView or TreeViewList | .pf-v5-c-form-control pf-m-search | Additional props for search input |
+## Testing
+
+At a minimum, a tree view should meet the following criteria:
+
+<List isPlain>
+  <ListItem>
+    <Checkbox id="treeview-a11y-checkbox-1" label={<span>The tree view has a unique accessible name provided via <code className="ws-code">aria-label</code> or <code className="ws-code">aria-labelledby</code>.</span>} description="If there are multiple tree views on a page, or other table elements, a unique accessible name makes it easier for users to distinguish one from the others. It also provides improved context if a user is navigating via rotor menu from any assistive technologies such as a screen reader." />
+  </ListItem>
+  <ListItem>
+    <Checkbox id="treeview-a11y-checkbox-2" label={<span>Only the root <code className="ws-code">ul</code> element within the tree view has the attribute <code className="ws-code">role="tree"</code>. Any nested <code className="ws-code">ul</code> elements instead have the attribute <code className="ws-code">role="group"</code></span>} />
+  </ListItem>
+  <ListItem>
+    <Checkbox id="treeview-a11y-checkbox-3" label="If there are more than 7 items at the root of the tree view, a typeahead search is provided." />
+  </ListItem>
+  <ListItem>
+    <Checkbox id="treeview-a11y-checkbox-4" label={<span>Within the context of the tree view, each tree view item can be navigated to and interacted with via keyboard. <kbd>Up Arrow</kbd> and <kbd>Down Arrow</kbd> will navigate to the next or previous item without expanding or collapsing any, respectively. <kbd>Left Arrow</kbd> will collapse an expanded item or navigate to its parent. <kbd>Right Arrow</kbd> will expand a collapsed item or navigate to its first child. <kbd>Enter</kbd> and <kbd>Space</kbd> will trigger a default action, typically selecting an item and/or expanding/collapsing it.</span>} />
+  </ListItem>
+  <ListItem>
+    <Checkbox id="treeview-a11y-checkbox-5" label={<span>When focus is outside of the tree view. <kbd>Tab</kbd> places focus on the first interactive item in the tree view or the last interactive element to have focus within the tree view. When focus is inside of the tree view, <kbd>Tab</kbd> places focus on the first focusable element outside of the tree view.</span>} />
+  </ListItem>
```

Comment:
This is made more true in the React PR I have up, but the more complicated examples may not work this way. It's trickier because a basic tree view should not have its items tabbable except when first entering the tree view, but things like separate selection/expansion, checkboxes, and action items complicate things.

Tabbing may be required in some of these instances because by default the Right and Left arrows have 2 roles: expanding/collapsing a parent node, and navigating to the first child/the parent when on a non-expandable or collapsed node. So using Right/Left arrow to get to a separate selection button, a checkbox, or an action item becomes difficult without omitting that default behavior, and if we do that it might be confusing that this behavior works on some tree views but not others.